src/scripts/browserDeviceProfile.js

```diff
@@ -294,12 +294,46 @@ import browser from './browser';
                     (browser.tizen && isTizenFhd ? 20000000 : null)));
     }
 
+    function getSpeakerCount() {
+        const AudioContext = window.AudioContext || window.webkitAudioContext || false; /* eslint-disable-line compat/compat */
+
+        if (AudioContext) {
+            const audioCtx = new AudioContext();
+
+            return audioCtx.destination.maxChannelCount;
+        }
+
+        return -1;
+    }
+
+    function getPhysicalAudioChannels(options) {
+        const numChannels = getSpeakerCount();
+        const isSurroundSoundSupportedBrowser = browser.safari || browser.chrome || browser.edgeChromium || browser.firefox || browser.tv || browser.ps4 || browser.xboxOne || false;
```

Comment:
I know this is copy-pasted from below, but I believe this covers pretty much all the browsers we target, so not sure how useful this is in practice.

Reply:
My solution determines the number of speakers the browser has available, so that users with stereo speakers don't get a 6 channel track by default when transcoding sources with surround sound. Previous logic was: if userSettings.allowedAudioChannels() or options.audioChannels are not set, use 6 channels unless isSurroundSoundSupportedBrowser == false

Reply:
What if we:
- remove `isSurroundSoundSupportedBrowser`
- use `allowedAudioChannels` if set
- use `numChannels` if `> 0`
- return 2 as default

So if user wants 7.1 on stereo speakers, let it be :upside_down_face: